### PR TITLE
Use coroutine instead of Rx when fetching offlineArea from local db

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #1581

Verified the changes for regression manually:
 * Opening the offline selector fragment
 * Ensuring that the size preview is shown correctly
 * Ensuring that I'm able to download offline area, view it in basemap on homescreen and delete it

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModel.kt [hunk 1]
```diff
@@ -87,7 +86,7 @@ constructor(
   fun initialize(args: OfflineAreaViewerFragmentArgs) {
     offlineAreaId = args.offlineAreaId
     viewModelScope.launch(ioDispatcher) {
-      val thisArea = offlineAreaRepository.getOfflineArea(offlineAreaId).await()
+      val thisArea = offlineAreaRepository.getOfflineArea(offlineAreaId)!!
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModel.kt:89]: Previously, await would return `null` if the `Maybe` was empty. Should we be doing the same here?

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModel.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModel.kt:89]: It was actually `Single`, so await would throw an exception

Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModel.kt:89]: Sorry, you're right!

**Review by gino-m — APPROVED** [review]
LGTM!

**Pull request merged**